spec/unit/request_registry_spec.rb

```diff
@@ -43,6 +43,25 @@
     it "should report number of times all matching pattern were requested" do
       expect(WebMock::RequestRegistry.instance.times_executed(WebMock::RequestPattern.new(:get, /.*example.*/))).to eq(3)
     end
+
+    describe "multithreading" do
+      let(:request_pattern) { WebMock::RequestPattern.new(:get, "www.example.com") }
+
+      # Reproduce a multithreading issue that causes a RuntimeError:
+      #   can't add a new key into hash during iteration.
+      it "works normally iterating on the requested signature hash while another thread is setting it" do
+        thread_injected = false
+        allow(request_pattern).to receive(:matches?).and_wrap_original do |m, *args|
```

Comment:
The idea of the spec here is that when `request_pattern.matches?` was called, we inject a second thread that writes to the same internal hash. It gives a deterministic results but it's coupled with the `times_executed` implementation. :man_shrugging: